ruby-monday-blog/spec/models/user_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:author) { User.new(first_name: 'First', last_name: 'Last', email: 'test@example.com', password: 'password', role: :author) }
+  let(:reader) { User.new(first_name: nil, last_name: nil, email: 'test@example.com', password: 'password', role: :reader) }
+
+  describe 'validations' do
+    context 'when user is an author' do
+      it 'validates presence of first name and last name' do
```

Comment:
@creativisionist I would try to also write a failing spec for the first name and last name such that all scenarios(passing as well as failing ones) are covered in my test cases. An example is given below -

```ruby
it "it is invalid with a blank first name" do
  person =  User.new(first_name: '', last_name: 'Last', email: 'test@example.com', password: 'password', role: :author)
  expect(person).to be_invalid
end
```

Also, it would be nice to break down the examples into invalid first_name, invalid last_name and both first_name and last_name being invalid . The reason behind this is a user could actually try out any of these scenarios.